### DISCUSSION TITLE
SessionManager as a preset manager

### DISCUSSION THREAD

**Discussion by trixon** [dlsc-software-consulting-gmbh/GemsFX#194] [Ideas]
Hi,
I've been happily using SessionManager for a while.
Now i finally found some time to work on "preset management" in Mapton and I got it working today.
I thought that I should continue to use SessionManager for the persistence part since I got that running already.

In order to get it working I had to add two things to SessionManager:
1. References to the listeners and "unregister" methods.
2. Store the value in the preferences while registering.

https://github.com/trixon/almond/blob/develop/util/src/main/java/com/dlsc/gemsfx/util/SessionManager2.java

Now when I want to save a preset I just create a new SessionManager for a new preferences node and after registering I call unregisterAll(). 
Recalling a preset does the same thing.

I also register a standard session manager that always stores and restores the settings, as usual.

Do you find this as is usable in the proper SessionManager and should I prepare a PR for it?

**Comment by dlemmermann** [comment]
A PR sounds good to me, yes.
